### ISSUE TITLE
[Package Request] - Python-3.10

### ISSUE THREAD

**Issue by tiadobatima** [amazonlinux/amazon-linux-2022#191]
**What package is missing from Amazon Linux 2022? Please describe and include package name.**
Python- 3.10

**Is this an update to existing package or new package request?**
Not quite an update. Ideally both 3.9 and 3.10 should be available.

**Is this package available in Amazon Linux 2?**
I looked for it, but couldn't find it on AL2

I'd like python 3.10 to be available along side with 3,9.

**Number of users impacted**
Not sure, but I imagine a lot of people would like to use the latest python.

**Comment by stewartsmith** [comment]
It is being worked on and should be available shortly.

**Comment by stewartsmith** [comment]
As of at least this release:
```
bash-5.1# rpm -qv system-release
system-release-2022.0.20221012-0.amzn2022.noarch
```

It's possible to install `python3.10`:
```
bash-5.1# dnf install -y python3.10 
Last metadata expiration check: 0:00:11 ago on Wed Oct 19 18:45:53 2022.
Dependencies resolved.
================================================================================
 Package             Arch       Version                   Repository       Size
================================================================================
Installing:
 python3.10          x86_64     3.10.4-1.amzn2022.0.2     amazonlinux      28 k
Installing dependencies:
 mpdecimal           x86_64     2.5.1-3.amzn2022.0.2      amazonlinux     101 k
 python3.10-libs     x86_64     3.10.4-1.amzn2022.0.2     amazonlinux     7.5 M

Transaction Summary
================================================================================
Install  3 Packages
..... <snipped for brevity>
```

and the default `/usr/bin/python3` remains as 3.9 for compatibility, and 3.10 is an option:
```
bash-5.1# ls -lah /usr/bin/pyth*
lrwxrwxrwx 1 root root   9 Sep 30 18:17 /usr/bin/python3 -> python3.9
-rwxr-xr-x 1 root root 16K Oct  2 15:17 /usr/bin/python3.10
-rwxr-xr-x 1 root root 16K Sep 30 18:17 /usr/bin/python3.9
```

```
bash-5.1# python3.10          
Python 3.10.4 (main, Oct  2 2022, 00:00:00) [GCC 11.3.1 20220421 (Red Hat 11.3.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> print("Hello World!")
Hello World!
```
So you can now use python 3.10 on AL2022!

**Comment by arwinlashawn** [comment]
Hi @stewartsmith, python3.10 does not seem to be available anymore on Amazon Linux 2023:

```
[ec2-user@ip-172-31-38-241 ~]$ sudo dnf install python3.10 -y
Last metadata expiration check: 0:06:42 ago on Wed Nov  1 05:50:41 2023.
No match for argument: python3.10
Error: Unable to find a match: python3.10
```
Do we have an ETA on when Python 3.10 will be supported on Amazon Linux 2023 via dnf? Thanks!

**Comment by stewartsmith** [comment]
> Hi @stewartsmith, python3.10 does not seem to be available anymore on Amazon Linux 2023:
> 
> ```
> [ec2-user@ip-172-31-38-241 ~]$ sudo dnf install python3.10 -y
> Last metadata expiration check: 0:06:42 ago on Wed Nov  1 05:50:41 2023.
> No match for argument: python3.10
> Error: Unable to find a match: python3.10
> ```
> 
> Do we have an ETA on when Python 3.10 will be supported on Amazon Linux 2023 via dnf? Thanks!

During preview, we replaced Python 3.10 with 3.11 - is there a specific need for 3.10 over 3.11?

**Comment by IvanMontillaM** [comment]
> During preview, we replaced Python 3.10 with 3.11 - is there a specific need for 3.10 over 3.11?

Hi @stewartsmith, some AI workloads for Stable Diffusion work better[0] with Python 3.10.

AI workloads with 3.11 are being worked on for stability and compatibility, but 3.10 is still better for it in the meantime.

--

[0]: The compatibility is much better for libraries like PyTorch, etc. Less issues/more frictionless overall.

**Comment by ozbenh** [comment]
While I have raised this internally, this comes at a cost for something that is EOL next year. Python 3.9 is our system python and isn't going away. We added 3.10 during the preview with the intend of providing a "newer" one for customers to use (but without a complete modules ecosystem). By the time we GA'd (which was delayed), 3.11 was current so we replaced it.

Recently, due to demand for newer python we added 3.12 (and will probably add 3.13). Adding 3.10 which is already on life support and will be gone next year doesn't seem that useful. Every python we add comes at a cost (CVEs).

Note that I had a look at it appears that RHEL 9 did exactly the same thing, with no 3.10 (3.9 is their system python and they provide 3.11 and 3.12).
